src/benchmarklib/benchmark_runner.hpp

```diff
@@ -42,63 +43,45 @@ class BenchmarkRunner {
 
-  // Calls _schedule_query if the scheduler is active, otherwise calls _execute_query and returns no tasks
-  std::vector<std::shared_ptr<AbstractTask>> _schedule_or_execute_query(const QueryID query_id,
-                                                                        const std::shared_ptr<SQLPipeline>& pipeline,
-                                                                        const std::function<void()>& done_callback);
-
-  // Schedule and return all tasks for named_query
-  std::vector<std::shared_ptr<AbstractTask>> _schedule_query(const QueryID query_id,
-                                                             const std::shared_ptr<SQLPipeline>& pipeline,
-                                                             const std::function<void()>& done_callback);
-
-  // Execute named_query
-  void _execute_query(const QueryID query_id, const std::shared_ptr<SQLPipeline>& pipeline,
-                      const std::function<void()>& done_callback);
-
-  // If visualization is enabled, stores an executed plan
-  void _store_plan(const QueryID query_id, SQLPipeline& pipeline);
+  // Schedules a run of the specified for execution. After execution, the result is updated. If the scheduler is
+  // disabled, the item is executed immediately.
+  void _schedule_item_run(const BenchmarkItemID item_id);
 
   // Create a report in roughly the same format as google benchmarks do when run with --benchmark_format=json
   void _create_report(std::ostream& stream) const;
 
-  std::shared_ptr<SQLPipeline> _build_sql_pipeline(const QueryID query_id) const;
-
-  struct QueryPlans final {
-    // std::vector<>s, since queries can contain multiple statements
-    std::vector<std::shared_ptr<AbstractLQPNode>> lqps;
-    std::vector<std::shared_ptr<AbstractOperator>> pqps;
-  };
-
-  // If visualization is enabled, this stores the LQP and PQP for each query. Its length is defined by the number of
-  // available queries.
-  std::vector<QueryPlans> _query_plans;
-
   const BenchmarkConfig _config;
 
-  std::unique_ptr<AbstractQueryGenerator> _query_generator;
+  std::unique_ptr<AbstractBenchmarkItemRunner> _benchmark_item_runner;
   std::unique_ptr<AbstractTableGenerator> _table_generator;
 
-  // Stores the results of the query executions. Its length is defined by the number of available queries.
-  std::vector<QueryBenchmarkResult> _query_results;
+  // Stores the results of the item executions. Its length is defined by the number of available items.
+  std::vector<BenchmarkItemResult> _results;
 
   nlohmann::json _context;
 
   std::optional<PerformanceWarningDisabler> _performance_warning_disabler;
 
   Duration _total_run_duration{};
+
+  // The atomic uints are modified by other threads when finishing an item, to keep track of when we can
+  // let a simulated client schedule the next item, as well as the total number of finished queries so far
+  std::atomic_uint _currently_running_clients{0};
+  std::atomic_uint _total_finished_runs{0};
```

Comment:
Like this?

Reply:
Hyrise uses the term "runs" to mean consecutive equal values (see RLE, JSM). Coming right out of the join operators, this confused me. Maybe "executions", "item_executions", or just "items"? 

comment still uses the term "queries" instead of "items"